spec/controllers/blocks_controller_spec.rb

```diff
@@ -0,0 +1,198 @@
+      get :index
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe "GET new" do
+    it "requires login" do
+      get :new
+      expect(response).to redirect_to(root_url)
+    end
+
+    it "succeeds" do
+      login
+      get :new
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe "POST create" do
+    it "requires login" do
+      post :create
+      expect(response).to redirect_to(root_url)
+    end
+
+    it "requires a blocked user" do
+      login
+      post :create
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+    end
+
+    it "requires valid variables" do
+      login
+      post :create, params: { block: { blocked_user_id: create(:user).id, hide_me: -1 } }
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+    end
+
+    it "requires an existing blocked user" do
+      login
+      post :create, params: { block: { blocked_user_id: ([1] + User.order(id: :desc).limit(1).pluck(:id)).max + 1 } }
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+      expect(assigns[:block].blocked_user).to be_nil
+    end
+
+    it "requires users not be the same" do
+      user = create(:user)
+      login_as(user)
+      post :create, params: { block: { blocked_user_id: user.id } }
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+      expect(flash[:error][:array]).to include("User cannot block themself")
+    end
+
+    it "succeeds" do
+      login
+      post :create, params: { block: { blocked_user_id: create(:user).id } }
+      expect(response).to redirect_to(blocks_url)
+    end
+
+    it "sets expected variables" do
+      blocker = create(:user)
+      blockee = create(:user)
+      login_as(blocker)
+      post :create, params: { block: { blocked_user_id: blockee.id, block_interactions: false, hide_them: Block::POSTS, hide_me: Block::ALL } }
+      expect(response).to redirect_to(blocks_url)
+      block = assigns(:block)
+      expect(block).not_to be_nil
+      expect(block.blocking_user).to eq(blocker)
+      expect(block.blocked_user).to eq(blockee)
+      expect(block.block_interactions).to eq(false)
+      expect(block.hide_them).to eq(Block::POSTS)
+      expect(block.hide_me).to eq(Block::ALL)
+    end
+  end
+
+  describe "GET edit" do
+    it "requires permission" do
+      block = create(:block)
+      login
+      get :edit, params: { id: block.id }
+      expect(response).to redirect_to(blocks_url)
+      expect(flash[:error]).to eq("You do not have permission to edit that block.")
```

Comment:
Might be worthwhile using the same error as below, for this, so people can't get data about how many blocks exist. (That is, respond "Block could not be found." if the user doesn't have permission to edit it.)